pallets/file-system/src/lib.rs

```diff
@@ -928,7 +932,7 @@ pub mod pallet {
             location: FileLocation<T>,
             fingerprint: Fingerprint<T>,
             size: StorageData<T>,
-            msp_id: ProviderIdFor<T>,
+            msp_id: Option<ProviderIdFor<T>>,
```

Comment:
You know, this makes me think that we don't really need this MSP argument here, having already passed the bucket. It's not like you can choose a different MSP than the one storing the bucket.

Worth setting up a task with @santikaplan to remove this argument.

Reply:
I agree. I also think it's worth merging the MSP param (`ProviderIdFor<T>`) with the value proposition id in a new struct so that where we use them both, we only have 1 `Option` instead of 2.

Reply:
Well, removing the MSP param would just leave the value prop param, so there you go.